scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/crypto/CryptoCodec.scala

```diff
@@ -117,16 +129,26 @@ object CryptoCodec {
         return -1
       }
 
-      if (!headerVerified) {
-        val (encryptedDataKey, initializationVector) = bigdlEncrypt.getHeader(in)
-        val dataKeyPlainText = conf.get(s"bigdl.read.dataKey.$encryptedDataKey.plainText")
-        bigdlEncrypt.init(cryptoMode, DECRYPT, dataKeyPlainText)
-        bigdlEncrypt.verifyHeader(initializationVector)
-        headerVerified = true
+      if (enableNativeAESCBC) {
+        // data file is encrypted by native AES cihper
+        // no Mac integrity check defaultly
+        val (_, initializationVector) = bigdlEncrypt.getHeader(in)
+        val ivStr = new String(initializationVector, StandardCharsets.UTF_8)
+        val dataKeyPlainText = conf.get(s"bigdl.read.dataKey.$ivStr.plainText")
+        bigdlEncrypt.asInstanceOf[BigDLAESCBCEncrypt]
+          .setInitializationVector(initializationVector)
+        bigdlEncrypt.init(AES_CBC_PKCS5PADDING, DECRYPT, dataKeyPlainText)
```

Comment:
you can add initializationVector as a param of `bigdlEncrypt.init`, and `verifyHeader` in `init`